src/mon/OSDMonitor.cc

```diff
@@ -11800,16 +11804,15 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
              prefix == "osd pool application disable" ||
              prefix == "osd pool application set" ||
              prefix == "osd pool application rm") {
-    err = prepare_command_pool_application(prefix, cmdmap, ss);
-    if (err == -EAGAIN)
+    bool modified = false;
+    err = prepare_command_pool_application(prefix, cmdmap, ss, &modified);
+    if (err == -EAGAIN) {
```

Comment:
the function doesn't appear to ever return `-EAGAIN`. I'm not sure if this check is meant as future proofing or not, but this is effectively (atm) a dead branch.